tests/integration/SlugTest.php

```diff
@@ -0,0 +1,140 @@
+            ->withFields([
+                'title' => [],
+            ])
+            ->withSlugAttributes([
+                'title',
+            ])
+            ->boot();
+    }
+
+    public function testBasicSlugModel(): void
+    {
+        $model = $this->module->getRepository()->create([
+            'title' => 'My title',
+            'slug' => ['en' => 'my-title'],
+        ]);
+
+        $this->assertEquals('my-title', $model->getSlug());
+    }
+
+    public function testBasicSlugModelDuplicate(): void
+    {
+        for ($i = 0; $i < 10; $i++) {
+            $model = $this->module->getRepository()->create([
+                'title' => 'My title',
+                'slug' => ['en' => 'my-title'],
+            ]);
+
+            $this->assertEquals($i === 0 ? 'my-title' : 'my-title-' . $i + 1, $model->getSlug());
+        }
+    }
+
+    public function testCanReuseSoftDeletedSlug(): void
+    {
+        $model = $this->module->getRepository()->create([
+            'title' => 'My title',
+            'slug' => ['en' => 'my-title'],
+        ]);
+
+        $this->assertCount(1, $model->slugs()->get());
+        $this->assertEquals('my-title', $model->getSlug());
+
+        $this->module->getRepository()->delete($model->id);
+
+        // Create a new model after the delete.
+        $newModel = $this->module->getRepository()->create([
+            'title' => 'My title',
+            'slug' => ['en' => 'my-title'],
+        ]);
+
+        $this->assertCount(1, $newModel->slugs()->get());
+        $this->assertEquals('my-title', $newModel->getSlug());
+
+        // Total slugs should be 2.
+        $this->assertCount(2, $this->module->getSlugModelClassName()::withTrashed()->get());
+
+        // Restore the deleted model.
+        $this->assertTrue($this->module->getRepository()->restore($model->id));
+
+        $model = $this->module->getModelClassName()::find($model->id);
+
+        $this->assertCount(1, $model->slugs()->get());
+        $this->assertEquals('my-title-2', $model->getSlug());
+    }
+
+    public function testCanReuseSoftDeletedSlugWithHistory(): void
+    {
+        $model = $this->module->getRepository()->create([
+            'title' => 'My title',
+            'slug' => ['en' => 'my-title'],
+        ]);
+
+        $model = $this->module->getRepository()->update($model->id, [
+            'slug' => ['en' => 'slug-update'],
+        ]);
+
+        $this->assertCount(2, $model->slugs()->get());
+        $this->assertEquals('slug-update', $model->getSlug());
+
+        $this->module->getRepository()->delete($model->id);
+
+        // Create a new model after the delete.
+        $newModel = $this->module->getRepository()->create([
+            'title' => 'My title',
+            'slug' => ['en' => 'slug-update'],
+        ]);
+
+        // Total slugs should be 4.
+        $this->assertCount(4, $this->module->getSlugModelClassName()::withTrashed()->get());
```

Comment:
Not sure if I'm missing something, but this should be 3, no?

- `$model` should have 2: `my-title` (inactive and soft-deleted) and `slug-update` (active and soft-deleted)
- `$newModel` should have 1: `slug-update` (active).

Reply:
You are correct, that is more optimal, this was due to the current flow, which I now have changed.